// @uiw/react-color/cjs/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Alpha: true,
  Compact: true,
  Chrome: true,
  Colorful: true,
  Circle: true,
  Block: true,
  EditableInput: true,
  EditableInputRGBA: true,
  EditableInputHSLA: true,
  Hue: true,
  Github: true,
  Material: true,
  Saturation: true,
  ShadeSlider: true,
  Sketch: true,
  Slider: true,
  Swatch: true,
  Wheel: true
};
Object.defineProperty(exports, "Alpha", {
  enumerable: true,
  get: function get() {
    return _reactColorAlpha["default"];
  }
});
Object.defineProperty(exports, "Block", {
  enumerable: true,
  get: function get() {
    return _reactColorBlock["default"];
  }
});
Object.defineProperty(exports, "Chrome", {
  enumerable: true,
  get: function get() {
    return _reactColorChrome["default"];
  }
});
Object.defineProperty(exports, "Circle", {
  enumerable: true,
  get: function get() {
    return _reactColorCircle["default"];
  }
});
Object.defineProperty(exports, "Colorful", {
  enumerable: true,
  get: function get() {
    return _reactColorColorful["default"];
  }
});
Object.defineProperty(exports, "Compact", {
  enumerable: true,
  get: function get() {
    return _reactColorCompact["default"];
  }
});
Object.defineProperty(exports, "EditableInput", {
  enumerable: true,
  get: function get() {
    return _reactColorEditableInput["default"];
  }
});
Object.defineProperty(exports, "EditableInputHSLA", {
  enumerable: true,
  get: function get() {
    return _reactColorEditableInputHsla["default"];
  }
});
Object.defineProperty(exports, "EditableInputRGBA", {
  enumerable: true,
  get: function get() {
    return _reactColorEditableInputRgba["default"];
  }
});
Object.defineProperty(exports, "Github", {
  enumerable: true,
  get: function get() {
    return _reactColorGithub["default"];
  }
});
Object.defineProperty(exports, "Hue", {
  enumerable: true,
  get: function get() {
    return _reactColorHue["default"];
  }
});
Object.defineProperty(exports, "Material", {
  enumerable: true,
  get: function get() {
    return _reactColorMaterial["default"];
  }
});
Object.defineProperty(exports, "Saturation", {
  enumerable: true,
  get: function get() {
    return _reactColorSaturation["default"];
  }
});
Object.defineProperty(exports, "ShadeSlider", {
  enumerable: true,
  get: function get() {
    return _reactColorShadeSlider["default"];
  }
});
Object.defineProperty(exports, "Sketch", {
  enumerable: true,
  get: function get() {
    return _reactColorSketch["default"];
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function get() {
    return _reactColorSlider["default"];
  }
});
Object.defineProperty(exports, "Swatch", {
  enumerable: true,
  get: function get() {
    return _reactColorSwatch["default"];
  }
});
Object.defineProperty(exports, "Wheel", {
  enumerable: true,
  get: function get() {
    return _reactColorWheel["default"];
  }
});

var _colorConvert = require("@uiw/color-convert");

Object.keys(_colorConvert).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _colorConvert[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _colorConvert[key];
    }
  });
});

var _reactColorAlpha = _interopRequireWildcard(require("@uiw/react-color-alpha"));

Object.keys(_reactColorAlpha).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorAlpha[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorAlpha[key];
    }
  });
});

var _reactColorCompact = _interopRequireWildcard(require("@uiw/react-color-compact"));

Object.keys(_reactColorCompact).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorCompact[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorCompact[key];
    }
  });
});

var _reactColorChrome = _interopRequireWildcard(require("@uiw/react-color-chrome"));

Object.keys(_reactColorChrome).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorChrome[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorChrome[key];
    }
  });
});

var _reactColorColorful = _interopRequireWildcard(require("@uiw/react-color-colorful"));

Object.keys(_reactColorColorful).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorColorful[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorColorful[key];
    }
  });
});

var _reactColorCircle = _interopRequireWildcard(require("@uiw/react-color-circle"));

Object.keys(_reactColorCircle).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorCircle[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorCircle[key];
    }
  });
});

var _reactColorBlock = _interopRequireWildcard(require("@uiw/react-color-block"));

Object.keys(_reactColorBlock).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorBlock[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorBlock[key];
    }
  });
});

var _reactColorEditableInput = _interopRequireWildcard(require("@uiw/react-color-editable-input"));

Object.keys(_reactColorEditableInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorEditableInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorEditableInput[key];
    }
  });
});

var _reactColorEditableInputRgba = _interopRequireWildcard(require("@uiw/react-color-editable-input-rgba"));

Object.keys(_reactColorEditableInputRgba).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorEditableInputRgba[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorEditableInputRgba[key];
    }
  });
});

var _reactColorEditableInputHsla = _interopRequireWildcard(require("@uiw/react-color-editable-input-hsla"));

Object.keys(_reactColorEditableInputHsla).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorEditableInputHsla[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorEditableInputHsla[key];
    }
  });
});

var _reactColorHue = _interopRequireWildcard(require("@uiw/react-color-hue"));

Object.keys(_reactColorHue).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorHue[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorHue[key];
    }
  });
});

var _reactColorGithub = _interopRequireWildcard(require("@uiw/react-color-github"));

Object.keys(_reactColorGithub).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorGithub[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorGithub[key];
    }
  });
});

var _reactColorMaterial = _interopRequireWildcard(require("@uiw/react-color-material"));

Object.keys(_reactColorMaterial).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorMaterial[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorMaterial[key];
    }
  });
});

var _reactColorSaturation = _interopRequireWildcard(require("@uiw/react-color-saturation"));

Object.keys(_reactColorSaturation).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorSaturation[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorSaturation[key];
    }
  });
});

var _reactColorShadeSlider = _interopRequireWildcard(require("@uiw/react-color-shade-slider"));

Object.keys(_reactColorShadeSlider).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorShadeSlider[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorShadeSlider[key];
    }
  });
});

var _reactColorSketch = _interopRequireWildcard(require("@uiw/react-color-sketch"));

Object.keys(_reactColorSketch).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorSketch[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorSketch[key];
    }
  });
});

var _reactColorSlider = _interopRequireWildcard(require("@uiw/react-color-slider"));

Object.keys(_reactColorSlider).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorSlider[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorSlider[key];
    }
  });
});

var _reactColorSwatch = _interopRequireWildcard(require("@uiw/react-color-swatch"));

Object.keys(_reactColorSwatch).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorSwatch[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorSwatch[key];
    }
  });
});

var _reactColorWheel = _interopRequireWildcard(require("@uiw/react-color-wheel"));

Object.keys(_reactColorWheel).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactColorWheel[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactColorWheel[key];
    }
  });
});
