// @uiw/react-color-compact/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _colorConvert = require("@uiw/color-convert");

var _reactColorEditableInput = _interopRequireDefault(require("@uiw/react-color-editable-input"));

var _reactColorEditableInputRgba = _interopRequireDefault(require("@uiw/react-color-editable-input-rgba"));

var _reactColorSwatch = _interopRequireDefault(require("@uiw/react-color-swatch"));

var _jsxRuntime = require("react/jsx-runtime");

var _excluded = ["prefixCls", "className", "style", "onChange", "color", "colors"];
var COLORS = ['#4D4D4D', '#999999', '#FFFFFF', '#F44E3B', '#FE9200', '#FCDC00', '#DBDF00', '#A4DD00', '#68CCCA', '#73D8FF', '#AEA1FF', '#FDA1FF', '#333333', '#808080', '#cccccc', '#D33115', '#E27300', '#FCC400', '#B0BC00', '#68BC00', '#16A5A5', '#009CE0', '#7B64FF', '#FA28FF', '#000000', '#666666', '#B3B3B3', '#9F0500', '#C45100', '#FB9E00', '#808900', '#194D33', '#0C797D', '#0062B1', '#653294', '#AB149E'];

function Point(props) {
  if (!props.checked) return null;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    style: {
      height: 5,
      width: 5,
      borderRadius: '50%',
      backgroundColor: (0, _colorConvert.getContrastingColor)(props.color)
    }
  });
}

var Compact = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  var _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'w-color-compact' : _props$prefixCls,
      className = props.className,
      style = props.style,
      onChange = props.onChange,
      color = props.color,
      _props$colors = props.colors,
      colors = _props$colors === void 0 ? COLORS : _props$colors,
      other = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var hsva = typeof color === 'string' && (0, _colorConvert.validHex)(color) ? (0, _colorConvert.hexToHsva)(color) : color;
  var hex = color ? (0, _colorConvert.hsvaToHex)(hsva).replace(/^#/, '') : '';
  var handleChangeCallback = (0, _react.useCallback)(function (hsv) {
    return onChange && onChange((0, _colorConvert.color)(hsv));
  }, []);

  var handleHex = function handleHex(value, evn) {
    if (typeof value === 'string' && (0, _colorConvert.validHex)(value) && /(3|6)/.test(String(value.length))) {
      handleChangeCallback((0, _colorConvert.hexToHsva)(value));
    }
  };

  var rgbProps = {
    style: {
      alignItems: 'baseline'
    },
    inputStyle: {
      boxShadow: 'none',
      backgroundColor: 'transparent',
      outline: 0
    }
  };
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    ref: ref,
    style: (0, _objectSpread2["default"])({
      background: '#f6f6f6',
      borderRadius: 3,
      display: 'flex',
      width: 240,
      flexWrap: 'wrap',
      paddingTop: 5,
      paddingLeft: 5
    }, style),
    className: [prefixCls, className || ''].filter(Boolean).join(' ')
  }, other), {}, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactColorSwatch["default"], {
      colors: colors,
      color: color ? (0, _colorConvert.hsvaToHex)(hsva) : undefined,
      rectProps: {
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(Point, {}),
        style: {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center'
        }
      },
      onChange: function onChange(hsvColor) {
        return handleChangeCallback(hsvColor);
      }
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      style: {
        display: 'flex',
        margin: '0 4px 3px 0'
      },
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_reactColorEditableInput["default"], {
        onChange: function onChange(evn, val) {
          return handleHex(val, evn);
        },
        labelStyle: {
          paddingRight: 5,
          marginTop: -1
        },
        value: hex.toLocaleUpperCase(),
        label: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          style: {
            width: 8,
            height: 8,
            backgroundColor: "#".concat(hex)
          }
        }),
        inputStyle: {
          outline: 'none',
          boxShadow: 'initial',
          background: 'transparent'
        },
        style: {
          flexDirection: 'row-reverse',
          flex: '1 1 0%',
          minWidth: 80
        }
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactColorEditableInputRgba["default"], {
        hsva: hsva,
        placement: "left",
        onChange: function onChange(result) {
          return handleChangeCallback(result.hsva);
        },
        aProps: false,
        rProps: rgbProps,
        gProps: rgbProps,
        bProps: rgbProps
      })]
    })]
  }));
});

Compact.displayName = 'Compact';
var _default = Compact;
exports["default"] = _default;
module.exports = exports.default;
