// @uiw/react-color-alpha/cjs/Pointer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Pointer = void 0;

var _react = _interopRequireWildcard(require("react"));

var _jsxRuntime = require("react/jsx-runtime");

var BOXSHADOW = 'rgb(0 0 0 / 37%) 0px 1px 4px 0px';

var Pointer = function Pointer(_ref) {
  var className = _ref.className,
      prefixCls = _ref.prefixCls,
      left = _ref.left,
      top = _ref.top;
  var style = {
    position: 'absolute',
    left: left,
    top: top
  };
  return (0, _react.useMemo)(function () {
    return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "".concat(prefixCls, "-pointer ").concat(className || ''),
      style: style,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: "".concat(prefixCls, "-fill"),
        style: {
          width: 18,
          height: 18,
          transform: left ? 'translate(-9px, -1px)' : 'translate(-1px, -9px)',
          boxShadow: BOXSHADOW,
          borderRadius: '50%',
          backgroundColor: 'rgb(248, 248, 248)'
        }
      })
    });
  }, [className, left, top, prefixCls]);
};

exports.Pointer = Pointer;
